evap/static/js/datagrid.js

```diff
@@ -0,0 +1,314 @@
+                    if (cell.is("[data-order]")) {
+                        data.orderValues.set(column, cell.data("order"));
+                    } else {
+                        data.orderValues.set(column, cell.html().trim());
+                    }
+                }
+                return data;
+            });
+    }
+
+    searchWordsOf(string) {
+        return string.toLowerCase().trim().split(/\s+/);
+    }
+
+    search(searchValue) {
+        this.state.search = searchValue;
+        this.filterRows();
+        this.update();
+    }
+
+    filter(filter) {
+        Object.assign(this.state.filter, filter);
+        this.filterRows();
+        this.update();
+    }
+
+    // Filters rows respecting the current search string and filters by their searchWords and filterValues
+    filterRows() {
+        const searchWords = this.searchWordsOf(this.state.search);
+        for (const row of this.rows) {
+            const isDisplayedBySearch = searchWords.every(searchWord => {
+                return row.searchWords.some(rowWord => rowWord.includes(searchWord));
+            });
+            const isDisplayedByFilters = Object.entries(this.state.filter).every(([name, filterValues]) => {
+                if (filterValues.length === 0) {
+                    return true;
+                } else {
+                    return filterValues.some(filterValue => {
+                        return row.filterValues.get(name).some(rowValue => rowValue === filterValue);
+                    });
+                }
+            });
+            row.isDisplayed = isDisplayedBySearch && isDisplayedByFilters;
+        }
+    }
+
+    sort(order) {
+        this.state.order = order;
+        this.sortRows();
+        this.update();
+    }
+
+    // Sorts rows respecting the current order by their orderValues
+    sortRows() {
+        for (const header of this.sortableHeaders.values()) {
+            header.removeClass("col-sorting-asc col-sorting-desc");
+        }
+        for (const [column, ordering] of this.state.order) {
+            this.sortableHeaders.get(column).addClass(`col-sorting-${ordering}`);
+        }
+
+        this.rows.sort((a, b) => {
+            for (const [column, order] of this.state.order) {
+                if (a.orderValues.get(column) < b.orderValues.get(column)) {
+                    return order === "asc" ? -1 : 1;
+                } else if (a.orderValues.get(column) > b.orderValues.get(column)) {
+                    return order === "asc" ? 1 : -1;
+                }
+            }
+            return 0;
+        });
+    }
+
+    // Reflects changes to the rows to the DOM
+    update() {
+        this.container.children().detach();
+        const elements = this.rows
+            .filter(row => row.isDisplayed)
+            .map(row => row.element);
+        this.container.append($(elements));
+        this.saveToStorage();
+    }
+
+    loadFromStorage() {
```

Comment:
maybe call this restoreStateFromStorage? so it's clear that this is about the state and that it will restore things, and not just load them (at least to me "restore" sounds like it would do more things (like applying the state) than just "load")

(i'd call the other method `saveStateToStorage` then accordingly)